mermithid/processors/misc/ComplexLineShape.py

```diff
@@ -0,0 +1,598 @@
+    # number of entries and correct number of points in the SELA, and if not, it generates a fresh file.
+    # When the variable regenerate is set as True, it generates a fresh file   
+    def check_existence_of_scatter_file(self, regenerate = False):
+        gases = self.gases
+        if regenerate == True:
+            print('generate fresh scatter file')
+            self.generate_scatter_convolution_file()
+        else: 
+            stuff_in_dir = os.listdir(self.path_to_osc_strengths_files)
+            if 'scatter_spectra_file' not in stuff_in_dir:
+                print('Scatter spectra folder not found, generating')
+                os.mkdir(self.path_to_osc_strengths_files+'scatter_spectra_file')
+                time.sleep(2)
+                self.generate_scatter_convolution_file()
+            else:
+                directory = os.listdir(self.path_to_osc_strengths_files + "/scatter_spectra_file")
+                strippeddirs = [s.strip('\n') for s in directory]
+                if 'scatter_spectra.npy' not in strippeddirs:
+                    self.generate_scatter_convolution_file()
+                test_file = self.path_to_osc_strengths_files+'scatter_spectra_file/scatter_spectra.npy' 
+                test_dict = np.load(test_file, allow_pickle = True)
+                if list(test_dict.item().keys())[0] != '{}_{}'.format(gases[0], gases[1]):
+                    print('first entry not matching, generating fresh files')
+                    self.generate_scatter_convolution_file()
+                elif len(test_dict.item()['{}_{}'.format(gases[0], gases[1])]) \
+                            != (1 + self.max_scatters+1)*(self.max_scatters+1)/2 - 1:
+                    print('Number of scatters not matching, generating fresh files')
+                elif len(test_dict.item()['{}_{}'.format(gases[0], gases[1])]['01_00'])\
+                != self.num_points_in_std_array:
+                    print('Binning do not match standard array, generating fresh files')
+                    self.generate_scatter_convolution_file()
+        return
+
+    # Given a function evaluated on the SELA, convolves it with a gaussian
+    def convolve_gaussian(self, func_to_convolve,gauss_FWHM_eV):
+        sigma = gaussian_FWHM_to_sigma(gauss_FWHM_eV)
+        resolution_f = self.std_gaussian(sigma)
+        ans = sp.signal.convolve(resolution_f,func_to_convolve,mode='same')
+        ans_normed = self.normalize(ans)
+        return ans_normed
+
+
+    def make_spectrum(self, gauss_FWHM_eV, prob_parameter, scatter_proportion, emitted_peak='shake'):
+        gases = self.gases
+        max_scatters = self.max_scatters
+        max_comprehensive_scatters = self.max_comprehensive_scatters
+        current_path = self.path_to_osc_strengths_files
+        # check_existence_of_scatter_files()
+        #filenames = list_files('scatter_spectra_files')
+        p = scatter_proportion
+        q = 1 - p
+        scatter_spectra = np.load(
```

Comment:
i think what Id prefer here is you define a variable for the filename on one line, and then call `load` on the next line. there are a few cases like this: this would be easier to read